subprojects/core/src/main/java/org/gradle/execution/taskgraph/CalculateTaskGraphDescriptor.java

```diff
@@ -17,17 +17,20 @@
 package org.gradle.execution.taskgraph;
 
 import org.gradle.TaskExecutionRequest;
+import org.gradle.internal.progress.BuildOperationDetails;
+import org.gradle.internal.progress.OperationFinishEvent;
 
 import java.util.List;
 import java.util.Set;
 
 /**
- * Operation Descriptor for Calculating the task graph of a build.
+ * Details about the task graph of a build.
+ *
  * This class is intentionally internal and consumed by the build scan plugin.
  *
  * @since 4.0
  */
-public class CalculateTaskGraphDescriptor {
+public final class CalculateTaskGraphDescriptor implements BuildOperationDetails<CalculateTaskGraphOperationResult> {
```

Comment:
We should probably rename this to CalculateTaskGraphDetails? I can make this as part of my story...